File: Part1_강의자료/Ch06_누적합/문제별코드/19951_태상이의훈련소생활/19951.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, M;
    scanf("%d %d", &N, &M);

    vector<int> arr(N + 1), acc(N + 1);
    for (int i = 1; i <= N; i++)
        scanf("%d", &arr[i]);
    
    int a, b, k;
    while (M--)
    {
        scanf("%d %d %d", &a, &b, &k);
        acc[a] += k;
        acc[b + 1] -= k;
    }

    int sum = 0;
    for (int i = 1; i <= N; i++)
    {
        sum += acc[i];
        printf("%d ", arr[i] + sum);
    }
    printf("\n");
    return 0;
}

File: Part1_강의자료/Ch06_누적합/문제별코드/19951_태상이의훈련소생활/19951.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
li = [[0] * (N + 1)] + list(map(int, input().split()))
acc = [0] * (N + 2)

for _ in range(M):
    a, b, k = map(int, input().split())
    acc[a] += k
    acc[b + 1] -= k

sum = 0
for i in range(1, N + 1):
    sum += acc[i]
    print(li[i] + sum, end=' ')
